erts/emulator/beam/erl_db_util.c

```diff
@@ -2349,8 +2349,8 @@ Eterm db_prog_match(Process *c_p,
 	    *esp++ = t;
 	    break;
 	case matchCall1:
-	    bif = (Eterm (*)(Process*, ...)) *pc++;
-	    t = (*bif)(build_proc, esp-1);
+	    bif = (BIF_RETTYPE (*)(BIF_ALIST)) *pc++;
+	    t = (*bif)(build_proc, esp-1, NULL);
```

Comment:
I think I've got everything updated. The one thing I'm not entirely sure of is if this _should_ be copied into `bif_args`, if this is an optimization for this _specific_ case, or something else?  I'm new to `c`; so, if I'm understanding this correctly, `esp-1` works _here_ without copying into `bif_args` because `esp-1` and `bif_args[0]` essentially point to the same value and both are pointers so semantically it's all good? Then by not copying into `bif_args[0]` we save an allocation/step?

And, please feel free to tell me to just _google_ more 😄 !

Thank you both for your time reviewing this, and please let me know if there's anything else I can change, improve, or have missed 😅  here.

Reply:
The copying of the arguments to `bif_args` when there are 2 or 3 arguments is only necessary because the arguments would be in the wrong order if we would pass `esp-2` or `esp-3`, respectively. When there is a single argument, there is no need to copy it because it can't be in the wrong order.

Reply:
Thank you @bjorng